src/test/ci/test_tested_versions.py

```diff
@@ -0,0 +1,150 @@
+                parse_version("0.0.3"),
+            ]
+        )
+    ) == ["0.0.1~0.0.3"]
+
+    # Simple range break within same minor
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("!0.0.3"),
+                parse_version("0.0.4"),
+                parse_version("0.0.5"),
+            ]
+        )
+    ) == ["0.0.1~0.0.2", "0.0.4~0.0.5"]
+
+    # Simple range break within with minor jump
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("!0.0.3"),
+                parse_version("0.0.4"),
+                parse_version("0.1.1"),
+            ]
+        )
+    ) == ["0.0.1~0.0.2", "0.0.4~0.1.1"]
+
+    # Simple range break within with major jump
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("!0.0.3"),
+                parse_version("0.0.4"),
+                parse_version("1.0.0"),
+            ]
+        )
+    ) == ["0.0.1~0.0.2", "0.0.4", "1.0.0"]
+
+    # Test we skip correctly unsupported first version
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("!0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("0.0.3"),
+            ]
+        )
+    ) == ["0.0.2~0.0.3"]
+
+    # Test we skip correctly unsupported multiple first versions
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("!0.0.1"),
+                parse_version("!0.0.2"),
+                parse_version("0.0.3"),
+            ]
+        )
+    ) == ["0.0.3"]
+
+    # Test we skip correctly unsupported last version
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("!0.0.3"),
+            ]
+        )
+    ) == ["0.0.1~0.0.2"]
+
+    # Test we skip correctly unsupported last versions
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("0.0.1"),
+                parse_version("!0.0.2"),
+                parse_version("!0.0.3"),
+            ]
+        )
+    ) == ["0.0.1"]
+
+    assert _get_supported_version_ranges(
+        TestedVersions(
+            [
+                parse_version("!0.0.1"),
+                parse_version("0.0.2"),
+                parse_version("0.0.3"),
+                parse_version("0.1.0"),
+                parse_version("!0.1.1"),
+                parse_version("1.0.1"),
+            ]
+        )
+    ) == ["0.0.2~0.1.0", "1.0.1"]
```

Comment:
Can you add a test with Non-semantic versions?

Reply:
Done in 6b3f68136a98a941c421ca399cf9c8a38678d580